src/gpu/intel/jit/dsl/dsl.cpp

```diff
@@ -349,7 +349,7 @@ tensor_t def(const std::string &name, layout_t layout, const expr_t &value,
         type::attr_t attr) {
     if (any(attr & type::attr_t::slm)) {
         gpu_assert(value.is_empty());
-        auto alloc_elems = into<int>(size_bytes(layout) / layout.type().size());
+        auto alloc_elems = into<int>(layout.elems());
```

Comment:
This is not correct for strided layouts, right? We either need an assertion somewhere in the DSL to check that any allocated SLM layout is dense (do we have it already?) or use `size_bytes()` style calculation.

Reply:
Good point, this had slipped my mind while I was refactoring it.